cmd/argo/commands/completion.go

```diff
@@ -136,20 +138,32 @@ variable.
 			shell := args[0]
 			rootCommand := NewCommand()
 			rootCommand.BashCompletionFunction = bashCompletionFunc
-			availableCompletions := map[string]func(io.Writer) error{
-				"bash": rootCommand.GenBashCompletion,
-				"zsh":  rootCommand.GenZshCompletion,
+			availableCompletions := map[string]func(out io.Writer, argo *cobra.Command) error{
+				"bash": runCompletionBash,
+				"zsh":  runCompletionZsh,
+				"fish": runCompletionFish,
 			}
 			completion, ok := availableCompletions[shell]
 			if !ok {
 				fmt.Printf("Invalid shell '%s'. The supported shells are bash and zsh.\n", shell)
 				os.Exit(1)
 			}
-			if err := completion(os.Stdout); err != nil {
+			if err := completion(os.Stdout, rootCommand); err != nil {
 				log.Fatal(err)
 			}
 		},
 	}
-
 	return command
 }
+
+func runCompletionBash(out io.Writer, argo *cobra.Command) error {
```

Comment:
sorry for the nitpicking, I think this repo typically refers to `*cobra.Command` parameter using `cmd`
so that global code search is more consistent
could you change this please.